signing_service/signing_service/tests/tests.py

```diff
@@ -116,6 +139,67 @@ def test_that_signing_service_should_reraise_different_socket_erros(self):
         mock_socket_close.assert_called_once()
 
 
+class SigningServiceHandleConnectionTestCase(TestCase):
+
+    def setUp(self):
+        self.host = '127.0.0.1'
+        self.port = 8000
+        self.initial_reconnect_delay = 2
+
+    def test_that__handle_connection_should_send_wrapped_service_refuse_if_frame_is_invalid(self):
+        # Prepare message with wrong signature
+        middleman_message = GolemMessageFrame(Ping(), 99)
+        raw_message = append_frame_separator(
+            escape_encode_raw_message(
+                middleman_message.serialize(
+                    private_key=CONCENT_PRIVATE_KEY,
+                )
+            )
+        )
+        malformed_raw_message = bytes(bytearray([max(raw_message[1] - 1, 0)])) + raw_message[1:]
```

Comment:
In case that `raw_message[0]==0` and `raw_message[1]==0`, `raw_message == malformed_message`, so test might be not deterministic